tests/celery_tests.py

```diff
@@ -132,34 +134,52 @@ def run_sql(
                 select_as_cta=cta,
                 tmp_table_name=tmp_table,
                 client_id=client_id,
+                cta_method=cta_method,
             ),
         )
         self.logout()
         return json.loads(resp.data)
 
-    def test_run_sync_query_dont_exist(self):
+    @parameterized.expand(
+        [CtaMethod.TABLE,]
+    )
+    def test_run_sync_query_dont_exist(self, cta_method):
         main_db = get_example_database()
         db_id = main_db.id
         sql_dont_exist = "SELECT name FROM table_dont_exist"
-        result1 = self.run_sql(db_id, sql_dont_exist, "1", cta=True)
-        self.assertTrue("error" in result1)
+        result = self.run_sql(
+            db_id, sql_dont_exist, f"1_{cta_method}", cta=True, cta_method=cta_method
+        )
+        if get_example_database().backend != "sqlite" and cta_method == CtaMethod.VIEW:
+            self.assertEqual(QueryStatus.SUCCESS, result["status"], msg=result)
+        else:
+            self.assertEqual(QueryStatus.FAILED, result["status"], msg=result)
 
-    def test_run_sync_query_cta(self):
+    @parameterized.expand([CtaMethod.TABLE, CtaMethod.VIEW])
```

Comment:
I like this pattern and frequently use it with `pytest` via `@pytest.mark.parametrize`. Unrelated to this change I wonder if we should consider using `pytest` as opposed to `unittest`.

Reply:
+1000
I would love to! 
Pytest supports unittests frameworks, so it would be easy to bootstrap the migration. Sounds like a really good topic for the next community meetup.

Reply:
Noted. I added this to the agenda.

Reply:
This pattern could really help clean up many of our tests 👍